open-craft/opencraft
# [BB-3785] Add Course Outline Page Preview

This PR adds the ability to preview the Course outline page in the new console.

**JIRA tickets**: https://tasks.opencraft.com/browse/BB-3785

**Dependencies**: None

**Screenshots**: 
<img width="831" alt="Screenshot 2021-03-21 at 6 22 51 PM" src="https://user-images.githubusercontent.com/1010244/111904766-bacf8880-8a72-11eb-93c0-7b1d2f1f438e.png">



**Testing instructions**:

1. Pull this PR on your local machine.
2. Change the following on the `registration/api/v2/serializers.py b/registration/api/v2/serializers.py` to avoid subdomain validation during registration.
```diff
diff --git a/registration/api/v2/serializers.py b/registration/api/v2/serializers.py
index 5e1f28d4..08ee02fa 100644
--- a/registration/api/v2/serializers.py
+++ b/registration/api/v2/serializers.py
@@ -362,8 +362,8 @@ class OpenEdXInstanceConfigSerializer(serializers.ModelSerializer):
         is_new_instance = self.instance is None
         is_changed = not is_new_instance and self.instance.subdomain != self.initial_data.get("subdomain")
 
-        if is_new_instance or is_changed:
-            validate_available_subdomain(value)
+        # if is_new_instance or is_changed:
+        #     validate_available_subdomain(value)
 
         return value
```
4. Perform a new registration or login on the console.
5. Go to ``http://localhost:3000/newconsole/theming/logos`` to see the new outline preview page.
**Reviewers**
